File: problem20.java
public class problem20 {
    // ** JAVA VERSION NOT WORKING **
    // Facotiral 100 is too large
    public static void main(String[] args) {
        long factorial100 = factorial(100L);
        System.out.println(factorial100);
        System.out.println(sumOfDigits(factorial100));
    }

    public static long factorial(long num){
        long product = 1L;
        for(long i = num; i > 0; i--){
            product *= i;
            System.out.println(product);
            System.out.println();
        }
        return product;
    }

    public static float sumOfDigits(float num){
        String numString = Integer.toString((int) num);
        float sum = 0;
        for(int i = 0; i < numString.length(); i++){
            sum += Integer.parseInt(numString.substring(i, i+1));
        }
        return sum;
    }
}

File: problem20.py
# Solved

import math
def main():
    factorial100 = list(str(math.factorial(100)))
    print(factorial100)
    print(sumOfDigits(factorial100))

def factorial(num):
    product = 1
    for i in range (num-1):
        product *= (i+1)
    return product

def sumOfDigits(num):
    # numString = str(num)
    # sum = 0
    # for i in range (num):
    #     print(i)
    #     sum += int(numString[i])
    # return sum
    b = [int(x) for x in num]
    return sum(b)



main()